versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/VersionStoreImpl.java

```diff
@@ -859,12 +862,13 @@ public PaginationIterator<Diff> getDiffs(
 
     ContentMapping contentMapping = new ContentMapping(persist);
 
-    Predicate<ContentKey> contentKeyPredicate = keyRestrictions.contentKeyPredicate();
+    ContentKeyPredicate contentKeyPredicate = keyRestrictions.contentKeyPredicate();
     Predicate<DiffEntry> keyPred =
         contentKeyPredicate != null
             ? d -> {
               ContentKey key = storeKeyToKey(d.key());
-              return key != null && contentKeyPredicate.test(key);
+              return key != null
+                  && contentKeyPredicate.test(key, () -> contentTypeForPayload(d.toPayload()));
```

Comment:
I am not sure this is correct. We are applying the filter to the "to" payload, why not apply it to the "from" payload as well? Payloads can change, if the same key is reused to create two objects of different content types. That's why I said elsewhere that this new filter should only ne used in `getKeys`, not in `getDiffs`.